Basic-Audio-Driver-Kotlin/app/src/main/java/com/tokbox/sample/basicaudiodriver/NoiseAudioDevice.kt

```diff
@@ -0,0 +1,157 @@
+package com.tokbox.sample.basicaudiodriver
+
+import android.content.Context
+import android.os.Environment
+import android.os.Handler
+import com.opentok.android.BaseAudioDevice
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.util.Random
+
+class NoiseAudioDevice(private val context: Context) : BaseAudioDevice() {
+    private val captureSettings: AudioSettings
+    private val rendererSettings: AudioSettings
+    private var capturerStarted: Boolean = false
+    private var rendererStarted: Boolean = false
+    private var audioDriverPaused: Boolean = false
+    private var capturerBuffer: ByteBuffer? = null
+    private var rendererBuffer: ByteBuffer? = null
+    private var rendererFile: File? = null
+    private var capturerHandler: Handler? = null
+    private val capturerIntervalMillis: Long = 1000
+
+    private val capturer: Runnable = object : Runnable {
+        override fun run() {
+            capturerBuffer?.rewind()
+            val rand = Random()
+            rand.nextBytes(capturerBuffer?.array())
+            audioBus.writeCaptureData(capturerBuffer, SAMPLING_RATE)
+            if (capturerStarted && !audioDriverPaused) {
+                capturerHandler?.postDelayed(this, capturerIntervalMillis)
+            }
+        }
+    }
+    private val rendererIntervalMillis: Long = 1000
+    private var rendererHandler: Handler? = null
+
+    private val renderer: Runnable = object : Runnable {
+        override fun run() {
+            rendererBuffer?.clear()
+            audioBus.readRenderData(rendererBuffer, SAMPLING_RATE)
+            try {
+                val stream = FileOutputStream(rendererFile)
+                stream.write(rendererBuffer?.array())
+                stream.close()
+            } catch (e: FileNotFoundException) {
+                e.printStackTrace()
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
```

Comment:
Maybe we should have a finally block to make sure stream is closed ?

Reply:
this is 1:1 Java code translation (so needs to be updated in Java code as well). 

It looks like `steram.close()` can also throw `IOException`, so this requires a bit more testing. I would leave this for now and unify this behavior later.

Reply:
agreed.